brave/brave-core
# News p3a usage time

Resolves brave/brave-browser#22945

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

For `Brave.Today.LastUsageTime`:

1) Start browser with fresh profile
2) Access local state page, verify that the metric does not exist in the p3a object
3) Use Brave News
4) Verify that the metric now exists and has a value of 1
5) Close browser, fast forward the time by one week, open browser, do not access news
6) Verify that the metric has a value of 2
7) Repeat step 5-6 with different time periods to test different metric values as needed.
8) Access News
9) Verify the metric has a value of 1

For `Brave.Today.DaysInMonthUsedCount`:

1) Start browser with fresh profile
2) Access local state page, verify that the metric does not exist in the p3a object
3) Use Brave News
4) Verify that the metric now exists and has a value of 1
5) Close browser, fast forward the time by one day, open browser, access news
6) Verify that the metric has a value of 2
7) Close browser, fast forward the time by one day, open browser, access news
8) Verify that the metric has a value of 3
9) Close browser, fast forward the time by 15 days, open browser, access news
10) Verify that the metric has a value of 3
11) Close browser, fast forward the time by 1 day, open browser, access news
12) Verify that the metric has a value of 3
13) Close browser, fast forward the time by 20 days, open browser, do not access news
14) Verify that the metric has a value of 2

For `Brave.Today.NewUserReturning`:

1) Start browser with fresh profile
2) Access local state page, verify that the metric exists in the p3a object and has a value of 0
3) Use Brave News
4) Verify that the metric now exists and has a value of 2
5) Close browser, fast forward the time by one day, open browser, access news
6) Verify that the metric has a value of 3
7) Close browser, fast forward the time by one week, open browser, access news
8) Verify that the metric has a value of 1
